File: Data Structures & Algorithms/house-robber/submission-1.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> dp(nums.size());

        int ans = 0;
        for (int i=0; i<nums.size(); i++) {
            dp[i] = max((i-1>= 0 ? dp[i-1] : 0), nums[i] + (i-2 >= 0 ? dp[i-2]: 0));
        }

        return dp[nums.size() - 1];
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const map = new Map();

        for (let i=0; i<nums.length; i++) {
            map.set(nums[i], true);
        }

        const dp = new Map();
        var ans = 0;
        for (let k of map.keys()) {
            var i = k;
            while (map.has(i) && !dp.has(i)) {
                i++;
            }
            console.log(k, )
            var c = dp.has(i) ? dp.get(i) : 0;
            while (k <= i) {
                dp.set(i, c);
                ans = Math.max(ans, c);
                c++;
                i--;
            }
        }

        return ans;
    }
}
